jujugui/static/gui/src/app/app.js

```diff
@@ -1320,6 +1238,9 @@ YUI.add('juju-gui', function(Y) {
           serviceName: 'charmstore'
         });
         this.set('charmstore', new Charmstore(charmstoreURL, apiPath, bakery));
+        // Store away the charmstore auth info.
+        var macaroon = bakery.getMacaroon();
```

Comment:
There's definitely opportunities to refine the code and architecture around auth. That said, I think it's more appropriate for subsequent cards, as this functionality will go through a lot of changes as we learn what works and doesn't when doing auth with macaroons.